Source/WebGPU/WebGPU/PresentationContextCoreAnimation.mm

```diff
@@ -27,30 +27,100 @@
 #import "PresentationContextCoreAnimation.h"
 
 #import "APIConversions.h"
-#import "Adapter.h"
+#import "Texture.h"
+#import <wtf/cocoa/TypeCastsCocoa.h>
 
 namespace WebGPU {
 
-PresentationContextCoreAnimation::PresentationContextCoreAnimation(const WGPUSurfaceDescriptor&)
+static CAMetalLayer *layerFromSurfaceDescriptor(const WGPUSurfaceDescriptor& descriptor)
+{
+    ASSERT(descriptor.nextInChain->sType == WGPUSType_SurfaceDescriptorFromMetalLayer);
+    ASSERT(!descriptor.nextInChain->next);
+    const auto& metalDescriptor = *reinterpret_cast<const WGPUSurfaceDescriptorFromMetalLayer*>(descriptor.nextInChain);
+    CAMetalLayer *layer = bridge_id_cast(metalDescriptor.layer);
+    return layer;
+}
+
+PresentationContextCoreAnimation::PresentationContextCoreAnimation(const WGPUSurfaceDescriptor& descriptor)
+    : m_layer(layerFromSurfaceDescriptor(descriptor))
 {
 }
 
 PresentationContextCoreAnimation::~PresentationContextCoreAnimation() = default;
 
-void PresentationContextCoreAnimation::configure(Device&, const WGPUSwapChainDescriptor&)
+void PresentationContextCoreAnimation::configure(Device& device, const WGPUSwapChainDescriptor& descriptor)
 {
-    // FIXME: Implement this.
+    m_configuration = std::nullopt;
+
+    if (descriptor.nextInChain)
+        return;
+
+    switch (descriptor.format) {
+    case WGPUTextureFormat_BGRA8Unorm:
+    case WGPUTextureFormat_BGRA8UnormSrgb:
+    case WGPUTextureFormat_RGBA16Float:
+        break;
+    case WGPUTextureFormat_RGB10A2Unorm:
```

Comment:
little unfortunate we do not have `BGR10A2` formats should we request that?

Reply:
(Conversation continued offline)